credoai/governance/governance.py

```diff
@@ -250,11 +220,79 @@ def export(self, filename=None):
         global_logger.info(export_status)
         return to_return
 
-    def _api_export(self, evidences):
+    def get_evidence_requirements(self, tags: dict = None):
+        """
+        Returns evidence requirements. Each evidence requirement can have optional tags
+        (a dictionary).
+
+        Parameters
+        ----------
+        tags : dict, optional
+            Tags to subset evidence requirements. If a model has been set, will default
+            to the model's tags. Evidence requirements will be returned that have no
+            tags or have the same tag as provided.
+
+        Returns
+        -------
+        List[EvidenceRequirement]
+        """
+        if tags is None:
+            tags = self.get_model_tags()
+
+        reqs = [
+            e for e in self._evidence_requirements if (not e.tags or e.tags == tags)
+        ]
+        return reqs
+
+    def get_evidence_tags(self):
+        """Return the unique tags used for all evidence requirements"""
+        return self._unique_tags
+
+    def get_model_tags(self):
+        """Get the tags for the associated model"""
+        if self._model:
+            return self._model["tags"]
+        else:
+            return None
+
+    def set_artifacts(self, model, training_dataset=None, assessment_dataset=None):
         global_logger.info(
-            f"Uploading {len(evidences)} evidences.. for use_case_id={self._use_case_id} policy_pack_id={self._policy_pack_id}"
+            f"Adding model ({model.name}) to governance. Model has tags: {model.tags}"
         )
-        self._api.create_assessment(self._use_case_id, self._policy_pack_id, evidences)
+        prepared_model = {"name": model.name, "tags": model.tags}
+        if training_dataset:
+            prepared_model["training_dataset_name"] = training_dataset.name
+        if assessment_dataset:
+            prepared_model["assessment_dataset_name"] = assessment_dataset.name
+        self._model = prepared_model
+
+    def set_evidence(self, evidences: List[Evidence]):
+        """
+        Update evidences
+        """
+        self._evidences = evidences
+
+    def tag_model(self, model):
```

Comment:
Can you add a docstring for this method?

> It has a lot of print statements, I think we should change them to `global_logger.info`

Never mind I didn't see it was interactive.